File: python/custom-set/custom_set.py
class CustomSet(object):
    def __init__(self, elements=[]):
        self.elements = []
        for element in elements:
            self.add(element)

    def isempty(self):
        return len(self.elements) == 0

    def __contains__(self, element):
        return element in self.elements

    def issubset(self, other):
        for element in self.elements:
            if element not in other.elements:
                return False
        return True

    def isdisjoint(self, other):
        for element in self.elements:
            if element in other.elements:
                return False
        return True

    def __eq__(self, other):
        if len(self.elements) != len(other.elements):
            return False
        else:
            for element in self.elements:
                if element not in other.elements:
                    return False
            return True

    def add(self, element):
        if element not in self.elements:
            self.elements.append(element)

    def intersection(self, other):
        result_elements = []
        for element in self.elements:
            if element in other.elements:
                result_elements.append(element)
        return CustomSet(result_elements)

    def __sub__(self, other):
        result_elements = []
        for element in self.elements:
            if element not in other.elements:
                result_elements.append(element)
        return CustomSet(result_elements)

    def __add__(self, other):
        result = CustomSet(self.elements)
        for element in other.elements:
            result.add(element)
        return result

File: kotlin/custom-set/src/main/kotlin/CustomSet.kt
class CustomSet(vararg ary: Int) {
    private val lst = mutableListOf<Int>()

    init {
        lst.addAll(ary.distinct())
    }

    fun isEmpty(): Boolean = lst.isEmpty()
    fun isSubset(other: CustomSet): Boolean = lst.all { other.contains(it) }
    fun isDisjoint(other: CustomSet): Boolean = isEmpty() || intersection(other).isEmpty()
    fun contains(other: Int): Boolean = lst.contains(other)
    fun intersection(other: CustomSet): CustomSet =
        CustomSet(*(lst.filter { other.sort().contains(it) }.toIntArray()))

    fun add(other: Int) {
        if (!lst.contains(other)) lst.add(other)
    }

    override fun equals(other: Any?): Boolean =
        if (other !is CustomSet) false else sort().toIntArray() contentEquals other.sort().toIntArray()

    operator fun plus(other: CustomSet): CustomSet = CustomSet(*((sort() + other.sort()).distinct().toIntArray()))
    operator fun minus(other: CustomSet): CustomSet = CustomSet(*((sort() - other.sort()).toIntArray()))
    private fun sort(): List<Int> = lst.toList().sorted()
}
